File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int sol = 0;
        while(left < right){
            int area = Math.min(height[left], height[right]) * (right - left);
            sol = Math.max(sol, area);
            if(height[left] < height[right]){
                left++;
            }else{
                right--;
            }
        }
        return sol;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        area = 0
        while left < right:
            w = right - left
            if height[left] <= height[right]:
                h = height[left] 
                left += 1
            else:
                h = height[right]
                right -= 1
            area = max(area, w * h)
        return area